### PR TITLE
Fix Debug of Unity Standalone Player

### PR DESCRIPTION
**Pull request by van800**
move UnityExe run-configuration from ReSharperHost and fix calling Debug on it.

### CODE REVIEW

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeDebugProfileState.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        console = createConsole(exeConfiguration.parameters.useExternalConsole, targetProcessHandler, commandLineString, executionEnvironment.project)
+        console.attachToProcess(targetProcessHandler)
+        targetProcessHandler.startNotify()
+
+        application.executeOnPooledThread {
+            UnityPlayerListener(project, {
+                if (!it.isEditor) {
+                    while (isAvailable(it.debuggerPort))
+                        Thread.sleep(1)
```
[the remark this answers is not in the document]
Comment (citizenmatt) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeDebugProfileState.kt:91]: Actually, this is going to have a couple of problems. Firstly, it's blocking the listener from continuing, and secondly, there's no guarantee that the player you've just got is the player you've just started. If there is more than one player broadcasting on the network, you'll get called multiple times. And it's possible you've got multiple standalone players running, too. So I'm not sure how you can correctly identify the player you've just started.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeDebugProfileState.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+                    UIUtil.invokeLaterIfNeeded {
+                        logger.trace("Connecting to Player with port: ${it.debuggerPort}")
+                        remoteConfiguration.port = it.debuggerPort
+                        result.setResult(createWorkerRunInfoFor(port, DebuggerWorkerPlatform.AnyCpu))
+                    }
+                }
+            }, {}, lifetime)
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeDebugProfileState.kt:99]: What is the duration of this `Lifetime`? It controls how long you're polling for - does it get terminated as soon as the result is set?

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.jetbrains.rider.plugins.unity.run.configurations.unityExe
+
+import com.intellij.execution.configurations.ConfigurationTypeBase
+import com.intellij.util.IconUtil
+import icons.UnityIcons
+
+class UnityExeConfigurationType : ConfigurationTypeBase("RunUnityExe", "Unity Executable",
+    "Unity executable configuration", IconUtil.scale(UnityIcons.RunConfigurations.UnityExe, null, 0.33f)) {
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt:8]: Perhaps these also need changing to "Unity Standalone Player". `RunUnityExe` looks like a persisted ID, worth checking and commenting that we shouldn't change it.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorProfileState.kt [hunk 4]
```diff
@@ -84,6 +84,18 @@ class UnityAttachToEditorProfileState(private val remoteConfiguration: UnityAtta
...
+                    // it is better because mono waits for attach, so rider would not miss any frames,
+                    // but if you disconnect and try to attach again it fails, because updatePidAndPort would try different port
+//                    val actualPort = com.jetbrains.rider.util.NetUtils.findFreePort(500013, setOf(port))
+//                    val processBuilder = ProcessBuilder(args)
+//                    processBuilder.environment().set("MONO_ARGUMENTS", "--debugger-agent=transport=dt_socket,address=${remoteConfiguration.address}:$actualPort,embedding=1,server=y,suspend=y")
+//                    val process = processBuilder.start()
+//
+//                    val actualPid = OSProcessUtil.getProcessID(process)
+//                    remoteConfiguration.pid = actualPid
+//                    remoteConfiguration.port = actualPort
```
[the remark this answers is not in the document]
Comment (van800) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/UnityAttachToEditorProfileState.kt:98]: Thank you for commenting on this, I just don't plan to fix it in this branch.

**Comment by van800** [comment]
During testing I found one problem.
when confuguration is cretaed https://github.com/JetBrains/resharper-unity/pull/1708/files#diff-6f3f3a372e24fe768e26f874998e2e38R64 configuration template also gets edited.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.jetbrains.rider.plugins.unity.run.configurations.unityExe
+
+import com.intellij.execution.configurations.ConfigurationTypeBase
+import com.intellij.util.IconUtil
+import icons.UnityIcons
+
+class UnityExeConfigurationType : ConfigurationTypeBase("RunUnityExe", "Standalone Player", // "RunUnityExe" preserved for compatibility
+    "Unity Standalone Player configuration", IconUtil.scale(UnityIcons.RunConfigurations.UnityExe, null, 0.33f)) {
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt:8]: Does this need to be scaled? We're not scaling the other icons

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfiguration.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+    override fun isNative(): Boolean {
+        return false
+    }
+
+    override fun getState(executor: Executor, environment: ExecutionEnvironment): RunProfileState{
+        val executorId = executor.id
+
+        if (executorId == DefaultDebugExecutor.EXECUTOR_ID)
+            return UnityExeDebugProfileState(this, DotNetRemoteConfiguration(project, ConfigurationTypeUtil.findConfigurationType(MonoRemoteConfigType::class.java).factory, name), environment)
```
Comment (citizenmatt) [inline rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfiguration.kt:26]: Should this be `MonoRemoteConfigType`? Should it be the `factory` passed in the constructor or `UnityExeConfigurationType`?

File: unity/EditorPlugin/PluginEntryPoint.cs [hunk 7]
```diff
@@ -333,6 +334,15 @@ private static void CreateProtocolAndAdvise(Lifetime lifetime, List<ProtocolInst
       }
     }
 
+    private static void GetBuildLocation(EditorPluginModel model)
+    {
+        var path = EditorUserBuildSettings.GetBuildLocation(EditorUserBuildSettings.selectedStandaloneTarget);
```
Comment (citizenmatt) [inline unity/EditorPlugin/PluginEntryPoint.cs:339]: What happens if a non-standalone player is selected, such as iPhone or Android?

Comment (van800) [reply unity/EditorPlugin/PluginEntryPoint.cs:339]: everything except non-standalone should be ignored, I haven't checked that though.

File: rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt (hunk 5 above)
Comment (van800) [reply rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/run/configurations/unityExe/UnityExeConfigurationType.kt:8]: omg, I missed that.

File: unity/EditorPlugin/PluginEntryPoint.cs (hunk 7 above)
Comment (van800) [reply unity/EditorPlugin/PluginEntryPoint.cs:339]: just checked. when android is selected - nothing happens, our standalone configuration is not generated, which is ok.

**Pull request merged**